test/Client.test.ts

```diff
@@ -266,6 +264,43 @@ describe('canMessageMultipleBatches', () => {
   })
 })
 
+describe('publishEnvelopes', () => {
+  it('can send a valid envelope', async () => {
+    const c = await newLocalHostClient()
+    const envelope = {
+      contentTopic: '/xmtp/0/foo/proto',
+      message: new TextEncoder().encode('hello world'),
+      timestamp: new Date(),
+    }
+    await c.publishEnvelopes([envelope])
+  })
+
+  it('rejects with invalid envelopes', async () => {
+    const c = await newLocalHostClient()
+    // Set a bogus authenticator so we can have failing tests
+    c.apiClient.setAuthenticator({
+      // @ts-ignore-next-line
+      createToken: async () => ({
+        toBase64: () => 'derp!',
+      }),
+    })
+    const envelope = {
+      contentTopic: buildUserContactTopic(c.address),
+      message: new TextEncoder().encode('hello world'),
+      timestamp: new Date(),
+    }
+
+    // `expect(promise).rejects.toThrow` fails but this works.
+    // Would love to understand why
+    try {
+      await c.publishEnvelopes([envelope])
```

Comment:
what about something like this?

```
expect(await c.publishEnvelopes([envelope])).rejects.toThrow(...));
```

i'm looking at this: https://jestjs.io/docs/expect#rejects

Reply:
In what you have above the `await` will cause the entire test function to abort when there is an error. The await is evaluated first, so the `expect` can't take over and catch the error.

I tried a bunch of different variations, but keep getting `Received function did not throw` test failures.

For example, this looks like something that would work (and does work correctly in many other tests)
```ts
await expect(c.publishEnvelopes([envelope])).rejects.toThrow()
```

Reply:
I figured it out...

The generated GRPC code doesn't throw actual JS errors. It just throws objects. Jest doesn't recognize that. If I wrap the error with `throw new Error(e)` in the publish code Jest picks up on it and the boring example I have above works fine.